src/components/action-popover/__internal__/action-popover-utils.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { ActionPopoverItem } from "../action-popover-item/action-popover-item.component";
+
+// Reusable type alias for item types
+type ReactItem = React.ReactChild | React.ReactFragment | React.ReactPortal;
+
+export const getItems = (
```

Comment:
comment (non-blocking): I wonder if this is the chance to decouple the implementation from specific react features like iterating over children and using DOM querying etc. If not at least this will isolate the parts that will need refactoring

Reply:
I'll certainly give it a shot. If it's straight forward, it's a nice thing to get done.